valohai_cli/commands/yaml/step.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Optional
+
+import click
+from valohai.internals.yaml import parse_config_from_source
+from valohai.yaml import config_to_yaml
+from valohai_yaml.objs import Config
+
+from valohai_cli.ctx import get_project
+from valohai_cli.messages import info
+from valohai_cli.models.project import Project
+
+
+@click.command()
+@click.argument('filenames', nargs=-1, type=click.Path(file_okay=True, exists=True, dir_okay=False), required=True)
+def step(filenames):
+    """
+    Update a step config(s) in valohai.yaml based on Python source file(s).
+
+    Example:
+
+        vh yaml step hello.py
+
+    :param filenames: Path(s) of the Python source code files.
+    """
+    project = get_project()
+    config_path = project.get_config_filename()
+
+    for source_path in filenames:
+        if not os.path.isfile(config_path):
+            update_yaml_from_source(source_path, project)
+            info("valohai.yaml generated.")
+        elif yaml_needs_update(source_path, project):
+            update_yaml_from_source(source_path, project)
+            info("valohai.yaml updated.")
+        else:
+            info("valohai.yaml already up-to-date.")
+
+
+def get_current_config(project: Project) -> Optional[Config]:
+    return project.get_config() if os.path.isfile(project.get_config_filename()) else None
```

Comment:
Wonder if this'd be better phrased as

```suggestion
def get_current_config(project: Project) -> Optional[Config]:
    try:
        return project.get_config()
    except FileNotFoundError:
        return None
```